Email Editor: Extract html processing helper from table class

allilevine opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

- Follow up to https://github.com/woocommerce/woocommerce/pull/60514
- Extract helpers for reuse in other blocks.
- Allow CSS variables in `sanitize_color()` ([reported here](https://github.com/woocommerce/woocommerce/pull/60514#discussion_r2314218854)).
- Fix double semicolon issues [reported here](https://github.com/woocommerce/woocommerce/pull/60514#discussion_r2314275322).
- Fix caption validation.

I chose `packages/php/email-editor/src/Integrations/Core/Renderer/Helpers/class-html-processing-helper.php` as the location to keep it close to where it's being used, but let me know if you'd prefer Utils.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Regression test adding and customizing a table block:
1. Go to WooCommerce > Settings > Advanced > Features and activate the Block Email editor feature.
2. Go to WooCommerce > Settings > Emails and open a transactional email editor and insert the table block.
3. Test customizing the table block: border styles, typography styles, fixed width, header and footer sections, backgrounds, padding.
4. Test adding a caption with a link.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

Email Editor: Extract html processing helper from table class for reuse in other block rendering classes.

</details>

allilevine committed:
Email Editor: Extract html processing helper from table class.

Commit:
Add changefile(s) from automation for the following project(s): packages/php/email-editor

allilevine committed:
Add CSS variable support and remove double semicolons.

allilevine committed:
Address coderabbit feedback.

allilevine committed:
Allow links in captions.

allilevine committed:
Remove color from caption css properties to prevent the background color from showing up as the text color.

Commit:
Add changefile(s) from automation for the following project(s): packages/php/email-editor, woocommerce

allilevine committed:
Revert "Remove color from caption css properties to prevent the background color from showing up as the text color."

This reverts commit 5b788ba44ab82a3da421cde13b4ab2e95077a5b4.

Commit:
Add changefile(s) from automation for the following project(s): packages/php/email-editor

allilevine committed:
Re-commit removing color from caption css properties.

allilevine committed:
Addres coderabbit feedback.

allilevine commented on packages/php/email-editor/src/Integrations/Core/Renderer/Helpers/class-html-processing-helper.php:
```diff
@@ -0,0 +1,330 @@
+		}
+
+		return $html->get_updated_html();
+	}
```
I've added removal with regex. `wp_kses` removes the entire caption.

triple0t reviewed (commented):
Looks good to me generally. 

> I chose packages/php/email-editor/src/Integrations/Core/Renderer/Helpers/class-html-processing-helper.php as the location to keep it close to where it's being used, but let me know if you'd prefer Utils.

Can you move the file to [packages/php/email-editor/src/Integrations/Utils](https://github.com/woocommerce/woocommerce/tree/trunk/packages/php/email-editor/src/Integrations/Utils)?

Can you also add a basic test for the `Html_Processing_Helper` class methods?

Thank you

PS: I'm approving in case this PR is blocking some other tasks, but I would appreciate it if you could implement my comments above.

allilevine committed:
Move file to Utils.

allilevine committed:
Add tests.

allilevine committed:
Fix css restrictive checks and regex error masking.

allilevine committed:
Remove the error logging.

allilevine committed:
Fix test expectation.

allilevine commented:
> Looks good to me generally.
> 
> > I chose packages/php/email-editor/src/Integrations/Core/Renderer/Helpers/class-html-processing-helper.php as the location to keep it close to where it's being used, but let me know if you'd prefer Utils.
> 
> Can you move the file to [packages/php/email-editor/src/Integrations/Utils](https://github.com/woocommerce/woocommerce/tree/trunk/packages/php/email-editor/src/Integrations/Utils)?
> 
> Can you also add a basic test for the `Html_Processing_Helper` class methods?
> 
> Thank you
> 
> PS: I'm approving in case this PR is blocking some other tasks, but I would appreciate it if you could implement my comments above.

@triple0t Thank you for the review! And your help with where to put this. I've moved it to Utils and added tests.
